src/estimagic/inference/shared.py

```diff
@@ -100,16 +110,17 @@ def calculate_inference_quantities(estimates, internal_estimates, free_cov, ci_l
 
     """
     if not isinstance(free_cov, pd.DataFrame):
-        free_index = np.array(internal_estimates.names)[internal_estimates.free_mask]
+        free_index = np.array(free_estimates.free_names)
         free_cov = pd.DataFrame(data=free_cov, columns=free_index, index=free_index)
+
     # ==================================================================================
     # Construct summary data frame for flat estimates
     # ==================================================================================
     registry = get_registry(extended=True)
 
-    df = pd.DataFrame(index=internal_estimates.names)
+    df = pd.DataFrame(index=free_estimates.all_names)
     df["value"] = tree_just_flatten(estimates, registry=registry)
-    df["free"] = internal_estimates.free_mask
+    df["free"] = free_estimates.free_mask
     df.loc[free_cov.index, "standard_error"] = np.sqrt(np.diag(free_cov))
 
     df["p_value"] = calculate_p_values(
```

Comment:
I suggest we use a different approach to building the DataFrame. 

1. Add `estimation_result` as argument. This can be a LieklihoodResult, MomentsResult or BootstrapResult
2. Instead of calculating stuff, call the `ci`, `se`, `p_values`, ....  methods of `estimation_result` 
3. Call `tree_just_flatten` on each of them
4. Combine them into a DataFrame
5. Continue as before

That way we get better re-use between bootstrap and the others and also more code sharing between the different formats in which you can get CIs, p_values, ...